src/main/java/com/cburch/logisim/std/arith/ShifterHdlGeneratorFactory.java

```diff
@@ -58,131 +57,100 @@ public LineBuffer getModuleFunctionality(Netlist TheNetlist, AttributeSet attrs)
             .pair("shiftMode", SHIFT_MODE_STRING);
     final var nrOfBits = attrs.getValue(StdAttr.WIDTH).getWidth();
     final var nrOfShiftBits = attrs.getValue(Shifter.SHIFT_BITS_ATTR);
+    contents.addRemarkBlock("""
+          ShifterMode represents when:
+          0 : Logical Shift Left
+          1 : Rotate Left
+          2 : Logical Shift Right
+          3 : Arithmetic Shift Right
+          4 : Rotate Right
+          """);
     if (Hdl.isVhdl()) {
-      contents.add("""
-            -----------------------------------------------------------------------------
-            --- ShifterMode represents when:                                          ---
-            --- 0 : Logical Shift Left                                                ---
-            --- 1 : Rotate Left                                                       ---
-            -----------------------------------------------------------------------------
-            --- 2 : Logical Shift Right                                               ---
-            --- 3 : Arithmetic Shift Right                                            ---
-            --- 4 : Rotate Right                                                      ---
-            -----------------------------------------------------------------------------
-            """)
-          .empty(2);
-
       if (nrOfBits == 1) {
-        contents.add("""
-            Result <= DataA WHEN {{shiftMode}} = 1 OR
-                                 {{shiftMode}} = 3 OR
-                                 {{shiftMode}} = 4 ELSE DataA AND NOT(ShiftAmount);
+        contents.addVhdlKeywords().add("""
+            result <= dataA {{when}} {{shiftMode}} = 1 {{or}}
+                                 {{shiftMode}} = 3 {{or}}
+                                 {{shiftMode}} = 4 {{else}} dataA {{and}} {{not}}(shiftAmount);
```

Comment:
wouldn't it be more readable to move {{else}} to next line and align with {{when}}?

Reply:
This is a coding convention question, I'll leave it as is for the moment until we decide on the VHDL coding convention.

Reply:
Main reason, I completely checked the generated VHDL and Verilog files on both correct alignment and correct functionality.
This was a very tedious task, and I propose to do any alternation to the current changes in separate PR's where one can go over the HDL-generators library by library and also implement the "post-replacement" property as proposed by @maehne . I know that all HDL-code generated by this PR does not have any more "bugs" (except maybe from the TTL library as I did not test those intensively)